protocol/message_persistence.go

```diff
@@ -1778,7 +1785,25 @@ func (db sqlitePersistence) DeleteMessages(ids []string) error {
 	}
 	inVector := strings.Repeat("?, ", len(ids)-1) + "?"
 
-	_, err := db.db.Exec("DELETE FROM user_messages WHERE id IN ("+inVector+")", idsArgs...) // nolint: gosec
+	tx, err := db.db.BeginTx(context.Background(), &sql.TxOptions{})
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err == nil {
+			err = tx.Commit()
+			return
+		}
+		// don't shadow original error
+		_ = tx.Rollback()
```

Comment:
Not a big deal, but would be nice to apply this pattern here: [Skip deferred error](https://zealous-polka-dc7.notion.site/Code-Smell-5-Skip-deferred-error-8e7ebcd4889549cbae51f5c20bb10bd7)

Reply:
Done. Let me know if I did it correctly